python/sparkdl/param/image_params.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Some parts are copied from pyspark.ml.param.shared and some are complementary
+to pyspark.ml.param. The copy is due to some useful pyspark fns/classes being
+private APIs.
+"""
+
+from pyspark.ml.param import Param, Params, TypeConverters
+from pyspark.sql.functions import udf
+
+from sparkdl.image.imageIO import imageArrayToStruct, imageSchema
+from sparkdl.param import SparkDLTypeConverters
+
+OUTPUT_MODES = ["vector", "image"]
+
+class HasInputImageNodeName(Params):
+
+    inputImageNodeName = Param(Params._dummy(), "inputImageNodeName",
+                               "name of the graph element/node corresponding to the input",
+                               typeConverter=TypeConverters.toString)
+
+    def __init__(self):
+        super(HasInputImageNodeName, self).__init__()
```

Comment:
I don't think you need this. A class will inherit its parent's methods.

Reply:
Haven't thought much about them as the MLlib params are created from a script (and these are created along the way). I will remove them.